project/src/ui/sketch_control/ControlPane.gd

```diff
@@ -132,6 +137,16 @@ func _ready():
 	if _board.get_sketch().is_compiled():
 		_built()
 
+func _on_edit_btn() -> void:
+	get_focus_owner().release_focus()
+	if (code_editor == null):
+		code_editor = code_main_window_t.instance()
+		code_editor.src_file = sketch_path
+		code_editor.sketch_owner = self
+		get_tree().root.add_child(code_editor)
```

Comment:
not sure if we want to add nodes to the tree root all the time? unless there is some good reason?